direct-auth/OktaDirectAuth/OktaDirectAuth/Onboarding/OnboardingManager.swift

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2021, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import UIKit
+import OktaIdx
+import OktaIdxAuth
+
+extension Notification.Name {
+    static let authenticationSuccessful = Notification.Name("AuthenticationSuccessful")
+    static let authenticationFailed = Notification.Name("AuthenticationFailed")
+}
+
+class OnboardingManager {
+    static let shared = OnboardingManager()
```

Comment:
If it is a singleton, why do we have `init(configuration:)`? Singleton must have a private constructor.

Reply:
Singletons don't actually need a private constructor; they can just represent the "default" state, but can have different instances as necessary. For example, `UserDefaults` or `FileManager` are useful Foundation classes that provide a shared singleton, yet can have custom instances constructed.